rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,418 @@
+
+#### Format
+
+The API generates expression matrices in three formats: `loom`, `mtx` and `csv`. From the [loom](http://loompy.org/) documentation page, the format is "an efficient file format for large omics datasets", enabling the storage of large expression matrices with metadata and also supports many popular data processing programming languages. For these reasons, `loom` is the service's default and most performant format. The `mtx` format adheres to the [10x feature-barcode matrix specification](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices) to enable compatibility with 10X methods from `scanpy` and `Seurat`. `csv` is supported as a raw text format for which performance may not be optimized.
+
+#### Feature
+
+Users are also able to specify the expression feature type, i.e. the matrix's axis label, as `gene` (default) or `transcript` names.
+
+#### Species (proposed)
+
+`species` is a potential fifth parameter that the endpoint may accept. Currently, `species` is not a top-level parameter and is rather specified within `filter` via the `cell_suspension.genus_species.ontology` and `cell_suspension.genus_species.ontology_label` filters. Since a distinct matrix is generated for each `species` requested, it may be a more intuitive user experience for it to be a top-level parameter. Additionally, it requires non-trivial logic for the Matrix Service to parse a complex `filter` object to determine which species are requested, i.e. which species to generate a matrix for. In order to simplify the user's mental model of the API as well as the service's logic complexity, `species` is proposed as a fifth request body parameter.
+
+Note: Currently, supported species are _Homo Sapiens_ and _Mus Musculus_.
+
+#### Response
+
+The response object returns a `request_id` that users can use to retrieve the status and results of a request by supplying it to the `GET` `/matrix` endpoint. A `status` and `message` field are also supplied to inform the user whether the request started successfully or not.
+
+*Note about multiple species:*
+If a request selects cells from multiple species, the Matrix Service will generate one matrix per species and thus return multiple `request_id`s. The convention here is for the human matrix to be reflected in `request_id` and all other species matrices to be reflected in a list of `request_id`s via `non_human_request_ids`.
+
+```
+{
+	'request_id': string,
+	'non_human_request_ids': list,
+	'status': one of ["In Progress", "Complete", "Failed", "Expired"],
```

Comment:
When would a request have the `expired` status?

Reply:
Following the 30 day `request_id` validity period. This is mentioned in _Availability_, but I'll add a link from here to there for clarity!